addOns/formhandler/src/main/java/org/zaproxy/zap/extension/formhandler/FormHandlerParam.java

```diff
@@ -98,18 +116,23 @@ protected void parseImpl() {
             }
         } catch (ConversionException e) {
             LOGGER.error("Error while loading key-value pair fields: {}", e.getMessage(), e);
-            this.fields = new ArrayList<>(DEFAULT_KEY_VALUE_PAIRS.size());
-            this.enabledFieldsNames = new ArrayList<>(DEFAULT_KEY_VALUE_PAIRS.size());
+            this.fields = new ArrayList<>(DEFAULT_FIELDS.size());
+            this.enabledFieldsNames = new ArrayList<>(DEFAULT_FIELDS.size());
         }
 
         if (this.fields.isEmpty()) {
-            // Grab the entry for every set in the map
-            for (Map.Entry<String, String> entry : DEFAULT_KEY_VALUE_PAIRS.entrySet()) {
-                // Store the key and value of that entry in variables
-                String name = entry.getKey();
-                String value = entry.getValue();
-                this.fields.add(new FormHandlerParamField(name, value));
-                this.enabledFieldsNames.add(name);
+            // Add all the default fields
+            this.fields.addAll(DEFAULT_FIELDS);
+            this.enabledFieldsNames =
+                    this.fields.stream()
+                            .map(FormHandlerParamField::getName)
+                            .collect(Collectors.toList());
```

Comment:
We could avoid iterating over the fields twice with the following (or alternatively use a loop?).
```java
this.enabledFieldsNames = 
    DEFAULT_FIELDS.stream()
         .peek(this.fields::add)
         .map(FormHandlerParamField::getName)
         .collect(Collectors.toList());
```

Reply:
Cool!

Reply:
Hmmm SonarLint seems to have an issue with using peek:

> "Stream.peek" should be used with caution (java:S3864)

> According to its JavaDocs, the intermediate Stream operation java.util.Stream.peek() “exists mainly to support debugging” purposes.

> A key difference with other intermediate Stream operations is that the Stream implementation is free to skip calls to peek() for optimization purpose. This can lead to peek() being unexpectedly called only for some or none of the elements in the Stream.

> As a consequence, relying on peek() without careful consideration can lead to error-prone code.

I'm happy to use your suggestion as long as we have agreement it's not an issue.

Reply:
@thc202 got any thoughts on this? Am I really better off looping twice vs using peek?

Reply:
The usage here is not a problem (as is now at least), you can also use a `map` that returns the own element but IMO this if statement should be removed, if the user does not want any we should not be adding them.

Reply:
This is the original behaviour just rejigged to use a list. I guess this would be hit if someone removed all the defaults and had a blank config and that’s your concern? How would I detect a fresh install and add them then?